web3/utils/events.py

```diff
@@ -147,11 +151,15 @@ def get_event_abi_types_for_decoding(event_inputs):
 def get_event_data(event_abi, log_entry):
     """
     Given an event ABI and a log entry for that event, return the decoded
+    event data
     """
     if event_abi['anonymous']:
         log_topics = log_entry['topics']
     else:
         log_topics = log_entry['topics'][1:]
+        raw_signature = next(iter(log_entry['topics']), None)
```

Comment:
Given that I found this line cryptic, I wanted to look at other ways to do this.

```python
# using cytoolz
from cytoolz import first, excepts

raw_signature = excepts(StopIteration, first, lambda v: None)(log_entry['topics'])

# more verbose
try:
    raw_signature = log_entry['topics'][0]
except IndexError:
    raw_signature = None

# or
if log_entry['topics']:
    raw_signature = log_entry['topics'][0]
else:
    raw_signature = None

# and probably my favorite because it produces better error messages
if not log_entry['topics']:
    raise MismatchedABI("Expected event to have a topic but it had not topics")  # this msg could be better
elif event_abi_to_log_topic(event_abi) != log_entry['topics'][0]:
    raise MismatchedABI("<put a good message here>")
```